packages/core/src/QueryManager.js

```diff
@@ -49,8 +89,9 @@ export class QueryManager {
       if (cache) {
         const cached = this.clientCache.get(sql);
         if (cached) {
+          const data = await cached;
           this._logger.debug('Cache');
-          result.fulfill(cached);
+          result.ready(type === 'exec' ? null : data);
```

Comment:
This line looks harmless, but raises a question: do we ever see exec calls hit the cache? I don't think we should... they should be issued with `cache: false | undefined`. And if we didn't shouldn't the cache result resolve to `undefined` anyway? So I don't think we need this conditional.

Reply:
Right, we don't cache exec requests but actually could benefit from it. Below, we add the promise for pending requests to the cache so we don't submit the same request twice.

This makes me think that we maybe want to have a separate map of pending requests that we put requests in no matter whether their results should be cached or not.

Reply:
I don’t think it is safe to cache exec queries, as they can modify DB state. For example if I add a table, drop it, then add it again (which could maybe happen for a data cube index table), I don’t want the last exec call ignored due to caching.

Reply:
Good example. A common request of creating an index would be nice to cache, though. I guess for now we can not cache and hopefully improve the parallelization when we have full query dependency tracking.